golang/internal/helper/docker/network_test.go

```diff
@@ -0,0 +1,116 @@
+//go:build unit
+// +build unit
+
+package docker_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/rs/zerolog/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"github.com/thanhpk/randstr"
+
+	dockerHelper "github.com/dyrector-io/dyrectorio/golang/internal/helper/docker"
+)
+
+type DockerNetworkHelperTestSuite struct {
+	suite.Suite
+	dockerClient client.Client
+	ctx          context.Context
+	prefix       string
+	networkNames []string
+}
+
+// Runs before everything
+func (testsuite *DockerNetworkHelperTestSuite) SetupSuite() {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		log.Fatal().Msg("Could not connect to docker socket.")
+	}
+	testsuite.dockerClient = *cli
+	testsuite.prefix = randstr.Hex(prefixLength)
+	testsuite.ctx = context.Background()
+	testsuite.networkNames = []string{"network1", "network2"}
+
+}
+
+// Runs after everything
+func (testsuite *DockerNetworkHelperTestSuite) TearDownSuite() {
+	testsuite.dockerClient.Close()
+}
+
+// this function executes before each test case
+func (testsuite *DockerNetworkHelperTestSuite) SetupTest() {
+	for i := range testsuite.networkNames {
```

Comment:
for each testcase all the networks are created, why?

Reply:
Consistency. So you can write test for manipulating those networks and expect those to be there by default.